src/components/AddHosts/utils.ts

```diff
@@ -1,27 +1,29 @@
 import { OcmClusterType } from './types';
+import { isSingleNodeCluster } from '../clusters/utils';
 
 export const getOpenshiftClusterId = (ocmCluster?: OcmClusterType) =>
   ocmCluster && ocmCluster.external_id;
 
 export const canAddHost = ({ cluster }: { cluster: OcmClusterType }) => {
-  if (getOpenshiftClusterId(cluster)) {
+  let isAllowed = false;
+
+  if (
+    getOpenshiftClusterId(cluster) &&
+    cluster.aiCluster &&
+    !isSingleNodeCluster(cluster.aiCluster)
+  ) {
     // Collision of "ready" states for AI and Telemetry
     // Show the tab only if telemetry reports "ready", not the AI
     if (
-      cluster.product?.id === 'OCP-AssistedInstall' &&
-      cluster.state === 'ready' &&
-      (!cluster.aiCluster || cluster.aiCluster.status === 'installed')
-    ) {
-      return true;
-    }
-    if (
-      cluster.product?.id !== 'OCP-AssistedInstall' &&
-      cluster.cloud_provider?.id === 'baremetal' &&
-      cluster.state === 'ready'
-    ) {
+      (cluster.state === 'ready' &&
+        cluster.product?.id === 'OCP-AssistedInstall' &&
+        (!cluster.aiCluster || cluster.aiCluster.status === 'installed')) ||
```

Comment:
This contradicts the condition above (`cluster.aiCluster`)

Reply:
Nice catch! 10x

Reply:
Fixed